### PR TITLE
fix: foxglove websocket player incorrect startTime

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
Fix the startTime behavior for foxglove websocket player when a connection is reset.

**Description**
The foxglove websocket player produces an incorrect _startTime_ (of 0) when a new session is started that uses "clock" time leading to incorrect plot displays like this: https://github.com/foxglove/studio/issues/7406

This behavior occurs because the state emit updates the `startTime` to the `currentTime` when the currentTime is less than the startTime:
https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L869

During a session reset, the currentTime is cleared and the #clockTime is cleared (https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L1290) so when the emit computes the currentTime it uses `ZERO_TIME` (https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L1186) to set startTime and never updates it again.

The reason this does _not_ happen for the first connection to the server is because topics are not yet available and the player emit bails early:
https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L853

On subsequent re-connections, topics are already available and the logic to set `startTime` runs before any clock message is received.

This PR updates the client `time` event handler to re-set the `startTime` and `endTime` on the first setting of `clockTime`. Since `clockTime` will be used for subsequent calls to `getCurrentTime` this results in a `startTime` that is properly updated to a server specified clock time instead of `ZERO_TIME`.

Fixes: #7406

### CODE REVIEW

File: packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts [hunk 1]
```diff
@@ -846,10 +854,17 @@ export default class FoxgloveWebSocketPlayer implements Player {
   // Potentially performance-sensitive; await can be expensive
   // eslint-disable-next-line @typescript-eslint/promise-function-async
   #emitState = debouncePromise(() => {
+    // When the connection closes, we should not emit anoyher state update because it will override the startTime?
```
Comment (snosenzo) [inline packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts:857]: unfinished comment w/ typo?

File: packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts [hunk 2]
```diff
@@ -846,10 +854,17 @@ export default class FoxgloveWebSocketPlayer implements Player {
...
+    //
+
     if (!this.#listener || this.#closed) {
       return Promise.resolve();
     }
 
+    // the reason this does not break us on startup is because this #topics gate
```
Comment (snosenzo) [inline packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts:864]: It feels like this comment is meant to be in the context of the PR? it's a bit confusing as to what "this" is referring to

**Review by snosenzo — APPROVED** [review]
Overall seems fine, but I'm wondering if it's worth emitting state without topics at all to avoid extra rerenders.

File: packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts (hunk 1 above)
Comment (defunctzombie) [reply packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts]: stale comment - removed

**Review by achim-k — APPROVED** [review]

**Comment by defunctzombie** [comment]
> but I'm wondering if it's worth emitting state without topics at all to avoid extra rerenders

We want to emit a state with a presence of "RECONNECTING" to show that feedback to the user.

**Pull request merged**